server/plugin/plugin.go

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package plugin
+
+import (
+	"math/rand"
+	gohttp "net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"text/template"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+
+	"github.com/mattermost/mattermost-plugin-solar-lottery/server/api"
+	"github.com/mattermost/mattermost-plugin-solar-lottery/server/api/solarlottery"
+	"github.com/mattermost/mattermost-plugin-solar-lottery/server/command"
+	"github.com/mattermost/mattermost-plugin-solar-lottery/server/config"
+	"github.com/mattermost/mattermost-plugin-solar-lottery/server/http"
+	"github.com/mattermost/mattermost-plugin-solar-lottery/server/store"
+	"github.com/mattermost/mattermost-plugin-solar-lottery/server/utils/bot"
+)
+
+type Plugin struct {
+	plugin.MattermostPlugin
+	configLock *sync.RWMutex
+	config     *config.Config
+
+	httpHandler *http.Handler
+	// notificationHandler api.NotificationHandler
+
+	Templates map[string]*template.Template
+}
+
+func NewWithConfig(conf *config.Config) *Plugin {
+	return &Plugin{
+		configLock: &sync.RWMutex{},
+		config:     conf,
+	}
+}
+
+func (p *Plugin) OnActivate() error {
+	botUserID, err := p.Helpers.EnsureBot(&model.Bot{
+		Username:    config.BotUserName,
+		DisplayName: config.BotDisplayName,
+		Description: config.BotDescription,
+	}, plugin.ProfileImagePath("assets/profile.png"))
+	if err != nil {
+		return errors.Wrap(err, "failed to ensure bot account")
+	}
+	p.config.BotUserID = botUserID
+
+	// Templates
+	bundlePath, err := p.API.GetBundlePath()
+	if err != nil {
+		return errors.Wrap(err, "couldn't get bundle path")
+	}
+	err = p.loadTemplates(bundlePath)
+	if err != nil {
+		return err
+	}
+
+	p.httpHandler = http.NewHandler()
+	// p.notificationHandler = api.NewNotificationHandler(p.newAPIConfig())
+
+	command.Register(p.API.RegisterCommand)
+
+	rand.Seed(time.Now().UnixNano())
```

Comment:
There's a separate ticket already to make it repeatable (a little non-trival since maps are involved). adding a --randomize would trigger randomization. Separate PR?